csr/include/sim/csr.hpp

```diff
@@ -0,0 +1,73 @@
+#ifndef INCL_CSR_HPP
+#define INCL_CSR_HPP
+
+#include <array>
+
+#include <sim/common.hpp>
+
+#include <sim/csr/idx.gen.hpp>
+#include <sim/csr/value.gen.hpp>
+
+namespace sim::csr {
+
+NODISCARD constexpr inline bool isCSRWritable(CSRIdx idx) noexcept {
+    constexpr bit::BitIdx HI = 11;
+    constexpr bit::BitIdx LO = 10;
+
+    constexpr uint16_t READ_ONLY_CODE = 0b11;
+
+    return bit::getBitField<uint16_t, HI, LO>(to_underlying(idx)) ==
+           READ_ONLY_CODE;
+}
+
+NODISCARD constexpr inline PrivLevel getCSRPrivLevel(CSRIdx idx) noexcept {
+    constexpr bit::BitIdx HI = 9;
+    constexpr bit::BitIdx LO = 8;
+
+    constexpr uint16_t HYPERVISOR_CODE = 0b10;
+
+    auto raw_priv_level =
+        bit::getBitField<uint16_t, HI, LO>(to_underlying(idx));
+
+    if (raw_priv_level == HYPERVISOR_CODE) {
+        return PrivLevel::MACHINE;
+    }
+
+    return PrivLevel{raw_priv_level};
+}
+
+struct CSRFile final {
+    enum class AccessStatus { OK, CSR_NOT_SUPPORTED };
```

Comment:
please add enum class directly to private or public part

Reply:
public access specifier is omitted in accordance with project codestyle